[35기 홍현진] Mission 2 : 모델링

hjz1010 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- users 앱 생성 및 User 테이블 생성
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- users App: 사용자와 관련된 기능을 구현할 앱
- User table: 사용자 정보 관리를 위한 클래스


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 이메일 validation을 함수화할 수는 없나요? (예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hjz1010 committed:
ADD : users 앱 생성 및 user 테이블 생성

- users App: 사용자와 관련된 기능을 구현할 앱
- User table: 사용자 정보 관리를 위한 클래스

Kang-kyunghun commented on students/hyeonjin/users/models.py:
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+
+# Create your models here.
+class User(models.Model): 
+    name         = models.CharField()
+    email        = models.CharField()
+    password     = models.CharField()
+    phone_number = models.CharField()
```
1. CharField는 max_length를 필수 인자로 받습니다. 넣어주세요.
2. email 필드에는 unique 속성을 추가하여 중복 저장을 방지해 주세요.
3. password의 max_length는 추후 암호화한 값이 들어갈 예정이기 때문에 200으로 해주세요.
4. 데이터의 이력 관리를 위해서 created_at, updated_at column을 추가해주세요.
5. makemigrations를 진행하여 문제없이 migrations 파일이 생성 되는지 확인하고 migration 파일도 같이 올려주세요.
6. Meta 속성으로 table 이름도 추가해 주세요.

```python
created_at = models.DateTimeField(auto_now_add=True)
updated_at = models.DateTimeField(auto_now=True)
```

hjz1010 committed:
JoinView 작성 진행중

hjz1010 committed:
Modify : models.py
- CharField에 필수인자 max_length 추가
- 중복 저장 방지를 위해 unique 속성 추가(unique=True)
- 데이터 이력 관리를 위해 created_at, updated_at 필드 추가
- migrate 진행

Jon9Beom commented on students/hyeonjin/users/views.py:
```diff
@@ -0,0 +1,26 @@
+# Create your views here.
+
+
+class JoinView(View):
```
`feature/hyeonjin-models` 에 맞지 않는 즉, 모델링과 관련되지 않은 코드들은 올라오지 않게 해주세요~~!

hjz1010 replied:
미션3을 진행하기 전에 새로운 branch를 만들고 view 작업을 해야하는 건가요? :)

Jon9Beom commented on students/hyeonjin/users/models.py:
```diff
@@ -0,0 +1,12 @@
+from django.db import models
+
+# Create your models here.
+
+
+class User(models.Model):
```
```suggestion
from django.db import models

class User(models.Model):
```

불필요한 공백 및 주석은 지워주세요~!

Jon9Beom commented on students/hyeonjin/users/models.py:
```diff
@@ -0,0 +1,12 @@
+    password = models.CharField(max_length=200)
+    phone_number = models.CharField(max_length=45)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```
Meta class를 통해 db_table 옵션으로 테이블 명을 복수로 작성해주세요!

Jon9Beom commented on students/hyeonjin/users/models.py:
```diff
@@ -0,0 +1,12 @@
+
+
+class User(models.Model):
+    name = models.CharField(max_length=45)
```
`=`을 기준으로 정렬해주세요!
가독성을 위한 컨벤션은 늘 지키려고 노력해주세요 :)

hjz1010 committed:
Modify : models.py
- 컨벤션에 맞지 않는 코드들 수정
- Meta class를 통해  데이터베이스 테이블 이름(복수) 추가

hjz1010 committed:
models.py(table 이름) 변경사항이 반영되도록 migrate 완료

Jon9Beom commented:
LGTM 🚀
